### PR TITLE
os: implement tempDirSync and use in tests.ts

### PR DESCRIPTION
**Pull request by choleraehyq**
Signed-off-by: Cholerae Hu <choleraehyq@gmail.com>

### CODE REVIEW

File: os.go [hunk 1]
```diff
@@ -193,3 +195,18 @@ func WriteFileSync(filename string, data []byte, perm uint32) []byte {
 	check(err)
 	return out
 }
+
+func TempDirSync(dirname string, prefix string) []byte {
+	name, err := afero.TempDir(fs, dirname, prefix)
```
Comment (qti3e) [inline os.go]: `fs.TempDir(fs, dirname, prefix)` ?
https://github.com/ry/deno/blob/7b5fbc71970056515c41f6983fc8e1f8ee443d11/os.go#L17

File: msg.proto [hunk 2]
```diff
@@ -99,4 +101,11 @@ message Msg {
   bytes write_file_sync_data = 131;
   uint32 write_file_sync_perm = 132;
   // write_file_sync_perm specified by https://godoc.org/os#FileMode
+
+  // TEMP_DIR_SYNC
+  string temp_dir_sync_dirname = 140;
+  string temo_dir_sync_prefix = 141;
```
Comment (starkwang) [inline msg.proto]: Ditto.

File: os.go [hunk 3]
```diff
@@ -46,6 +46,8 @@ func InitOS() {
 		case Msg_WRITE_FILE_SYNC:
 			return WriteFileSync(msg.WriteFileSyncFilename, msg.WriteFileSyncData,
 				msg.WriteFileSyncPerm)
+		case Msg_TEMP_DIR_SYNC:
+			return TempDirSync(msg.TempDirSyncDirname, msg.TemoDirSyncPrefix)
```
Comment (starkwang) [inline os.go]: Ditto. 😉

Comment (choleraehyq) [reply os.go]: ...My faults. Thank you so much :-)

**Comment by choleraehyq** [comment]
@ry Sorry, I accidentally closed this PR, can you reopen it?

**Pull request closed without merging**